### PR TITLE
Fix auto open filter menu

### LINKED ISSUE #9412: The "Add Filter" menu appears in the corner of the screen after close
What you were expecting:
The "Add Filter" menu should close (the status is open in false) after clicking on "Remove all filters" and button  "Add Filter" should disappear.

What happened instead:
The "Add Filter" button disappears, since there are no filters, but as soon as you enter a new filter, the "Add Filter" menu immediately appears in the corner of the screen.

Steps to reproduce:

1. Open demo https://marmelab.com/react-admin-helpdesk/#/tickets
2. Choose "Customer" filter
3. Click on "Add filter"
4. Click on "Remove all filters"
5. Choose again "Customer" filter
6. The "Add filter" menu appears in the corner of the screen.

https://github.com/marmelab/react-admin/assets/10139861/cf072491-597d-4fa7-9065-27d843a3e14d

Screenshot 2023-10-31 at 10 21 07

**Environment**

* React-admin version: 4.12.3
* React version: 18.2.0
* Browser: Chrome Version 118.0.5993.88 (Official Build) (x86_64)

### PR DESCRIPTION
**Pull request by erwanMarmelab**
## Problem
The "Add Filter" button disappears, since there are no filters, but as soon as you enter a new filter, the "Add Filter" menu immediately appears in the corner of the screen.
image
video

## Solution
Set state to close on "Remove all filters" click

## TODOs
- [x] fix bug
- [x] add a test

## Step to reproduce the bug
1. Open [storybook](http://localhost:9010/?path=/story/ra-ui-materialui-input-autocompletearrayinput--as-list-filter)
3. Choose "Author" filter
4. Click on "Add filter"
5. Click on "Remove all filters"
6. Choose again "Author" filter
7. The "Add filter" menu appears in the corner of the screen.

## Issue
Closes #9412

### CODE REVIEW

File: packages/ra-ui-materialui/src/list/filter/FilterButton.tsx [hunk 1]
```diff
@@ -210,7 +210,12 @@ export const FilterButton = (props: FilterButtonProps): JSX.Element => {
                     </MenuItem>
                 )}
                 {hasFilterValues && (
-                    <MenuItem onClick={() => setFilters({}, {}, false)}>
+                    <MenuItem
+                        onClick={() => {
+                            setFilters({}, {}, false);
+                            // setOpen(false);
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/filter/FilterButton.tsx:216]: commented out code

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.spec.tsx [hunk 2]
```diff
@@ -1102,4 +1103,38 @@ describe('<AutocompleteArrayInput />', () => {
         });
         expect(screen.getByDisplayValue('French')).not.toBeNull();
     });
+
+    it('should close the filter menu on removing all filters', async () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteArrayInput.spec.tsx:1107]: the test should be in FilterButton.spec.tsx

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.stories.tsx [hunk 3]
```diff
@@ -593,122 +593,3 @@ export const InsideReferenceArrayInputWithCreationSupport = () => (
         <Resource name="books" edit={BookEditWithReferenceAndCreationSupport} />
     </Admin>
 );
-
-const booksDataProvider = fakeRestDataProvider({
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteArrayInput.stories.tsx:597]: Isn't this code used for tests on the AutocompleteArrayInput?

Also, linter wqrnings.

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.stories.tsx (hunk 3 above)
Comment (erwanMarmelab) [reply packages/ra-ui-materialui/src/input/AutocompleteArrayInput.stories.tsx:597]: This data was only used for the test I made in [this commit](d02ad81ba29f90c155eae8ce23fd8c5b93819a55), but this test is now in `FisterButton.spec.tsx` and works with the `FisterButton.stories.tsx`

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**